certbot-ci/certbot_integration_tests/utils/acme_server.py

```diff
@@ -167,6 +176,14 @@ def _prepare_boulder_server(self):
         os.rename(join(instance_path, 'test/rate-limit-policies-b.yml'),
                   join(instance_path, 'test/rate-limit-policies.yml'))
 
+        if self._dns_server:
+            # Change Boulder config to use the provided DNS server
+            with open(join(instance_path, 'test/config/va.json'), 'r') as file_h:
```

Comment:
Oops, I should have piped up yesterday. Wasn't aware this was on the radar for merging :scream:.

At some point after this code was written, Boulder's `test/config` environment was updated to [make use of remote VAs](https://github.com/letsencrypt/boulder/blob/bcca7c2d05e2ced89f726adccd37ab388b6c9c9d/test/config/va.json#L32-L42). They are configured separately in `va-remote-{a,b}.json`.

As written, I think this will fail because both remote VAs will fail to query the right DNS server and the `EnforceMultiVA` feature flag is on as well.

I do not think it is urgent, as I am not planning to integrate Boulder with this flag yet anyway.

Reply:
Thanks for letting us know!

I agree it's not urgent, but think we should fix it or rip the boulder code out relatively soon. I think it's easy to do and I think it's better to fix the problem now rather than letting people stumble into it and rediscover it in the future. I created https://github.com/certbot/certbot/issues/8453 to track that.